ci/jobs/satellite6-pulp.yaml

```diff
@@ -0,0 +1,187 @@
+                nova keypair-add --pub-key pulp-jenkins.pub $KEY_NAME
+            fi
+            nova keypair-list
+
+            # Setup security group rules.
+            echo "Setup firewall rules that Satellite 6 needs to work properly."
+            SECURITY_GROUP_NAME=pulp-jenkins
+            SECURITY_GROUP_DESC="Security group for Satellite6-Pulp testing"
+            if [[ -z `nova secgroup-list | grep $SECURITY_GROUP_NAME` ]]; then
+                nova secgroup-create $SECURITY_GROUP_NAME "$SECURITY_GROUP_DESC"
+            fi
+            nova secgroup-list
+            if [[ -z `nova secgroup-list-rules $SECURITY_GROUP_NAME | grep "0.0.0.0/0"` ]]; then
+                nova secgroup-add-rule $SECURITY_GROUP_NAME icmp -1 -1 0.0.0.0/0
+                for tcp_port in 22 80 443 5000 5646 5647 5671 8000 8140 8443 9090; do
+                    nova secgroup-add-rule $SECURITY_GROUP_NAME tcp $tcp_port $tcp_port 0.0.0.0/0
+                done
+                for udp_port in 53 69; do
+                    nova secgroup-add-rule $SECURITY_GROUP_NAME udp $udp_port $udp_port 0.0.0.0/0
+                done
+            fi
+            nova secgroup-list-rules $SECURITY_GROUP_NAME
+
+            echo "Booting an OpenStack instance and getting its floating IP address..."
+            # Note: the flavor "c3.xlarge" might cause error on OS platform;
+            # Note: the flavor "c3.large" is stable but will likely cause memory
+            # error on Satellite6 installer job.
+            nova boot --flavor 'c3.xlarge' \
+                --image $IMAGE_ID \
+                --key-name $KEY_NAME \
+                --security-groups $SECURITY_GROUP_NAME \
+                $INSTANCE_NAME
+            INSTANCE_IP=
+            while [[ -z "${{INSTANCE_IP}}" ]]; do
+                echo "The network is not yet ready..."
+                sleep 1
+                INSTANCE_IP=`nova list | grep $INSTANCE_NAME | awk {{' print $13 '}} | head -1`
+            done
+            echo "The network is ready! The floating-IP of this instance is: $INSTANCE_IP"
+            sleep 10
+
+            INSTANCE_ID=`nova list | grep $INSTANCE_NAME | awk {{' print $2 '}} | head -1`
+
+            # Inject environment variables.
+            echo "Injecting environment variables..."
+            echo "KEY_NAME=${{KEY_NAME}}" > openstack.properties
+            echo "INSTANCE_IP=${{INSTANCE_IP}}" >> openstack.properties
+            echo "INSTANCE_ID=${{INSTANCE_ID}}" >> openstack.properties
+            echo "SECURITY_GROUP_NAME=${{SECURITY_GROUP_NAME}}" >> openstack.properties
```

Comment:
You can add all the required variables to be inject into one file. Since they will all leave on the environment and there will not be a separation between them.